kratos/utilities/atomic_utilities.h

```diff
@@ -0,0 +1,92 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//                   Denis Demidov
+//
+
+#if !defined(KRATOS_ATOMIC_UTILITIES_H_INCLUDED )
+#define  KRATOS_ATOMIC_UTILITIES_H_INCLUDED
+
+
+// System includes
+#include<omp.h>
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+/**
+ * collection of utilities for atomic updates of simple types. (essentially mimics the omp atomic)
+ */
+
+/** @param target variable being atomically updated by doing target += value
+ * @param value value being added
+ */ 
+template<class TDataType>
+inline void AtomicAdd(TDataType& target, const TDataType& value ) {
```

Comment:
```suggestion
inline void AtomicAdd(const TDataType& rValue, TDataType& rTarget) {
```

I think this would be more consistent

Reply:
i don't agree about this one. it is target += value so the order is correct...

Reply:
ok that makes sense

Still you might want to use the proper names according to the style guide to not get refactored at some point ;)